### ISSUE TITLE
EXE005 yields false positives for scripts using nix-shell in the shebang

### ISSUE THREAD

**Issue by SuperSandro2000** [astral-sh/ruff#27691]
### Summary

This is a valid nix-shell shebang to make a python script executable and bring all dependencies via nix, but it yields EXE005 which is a false positive here.

```python
#!/usr/bin/env nix-shell
#!nix-shell -I nixpkgs=./. -i python3 -p "requests" -p ruff
print("Hello world")
```

https://play.ruff.rs/2cc15e54-e74e-430c-a537-c6f6e2bc6761

### Version

0.16.1

**Comment by ntBre** [comment]
Initially I thought we'd have to special-case `nix-shell` to avoid this, but I guess another option is just not emitting the diagnostic if the file already has a valid shebang. That seems reasonable to me.

**Comment by kowanietz** [comment]
I'd like to work on this one.
